source/slang/slang-legalize-types.cpp

```diff
@@ -209,8 +209,8 @@ bool isPointerToResourceType(IRType* type)
 {
     while (auto ptrType = as<IRPtrTypeBase>(type))
     {
-        if (ptrType->getAddressSpace() == SpvStorageClassStorageBuffer ||
-            ptrType->getAddressSpace() == SpvStorageClassPhysicalStorageBufferEXT)
+        if (ptrType->getAddressSpace() == AddressSpace(SpvStorageClassStorageBuffer) ||
```

Comment:
I see the value for SpvStorageClassStorageBuffer is 12, so there is no corresponding address space value.
So can we remove this comparison as it will always be `false`?

Reply:
The comparison will not always be false, Pointers are assigned the address space of either `AddressSpace` or `SpvStorageClass*` as a value.

It is important to note that `AddressSpace` does not have equal values to equal-counter-parts in `SpvStorageClass*`

Reply:
Sorry, I don't understand your explanation here.

Do you mean 
`ptrType->getAddressSpace() ` could returns a value out of definition in
```
enum class AddressSpace
{...}
```
?

Reply:
That's correct, for now we are still using spv storage class enum values directly as the address space operand when legalizing spirv. We should stop doing that and introduce proper enum values to AddressSpace but that is out of the scope of this PR.